### PR TITLE
change device_tracker icon to reflect state

### PR DESCRIPTION
**Pull request by Mariusthvdb**
initial pr, need to add test for attribute `source_type: router`

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by Mariusthvdb** [comment]
please help me out adding the test for source_type.
```
  switch (stateObj?.attributes.source_type) {
    case "router":
      switch (state) {
        case "home":
          return mdiConnect;
        case "not_home":
          return mdiDisconnect;
        default:
          return mdiAccount;
      }
```
is what domain cover does, but not sure if that is ok here

**Comment by Mariusthvdb** [comment]
please let me know what is needed to make this go forward?

File: src/common/entity/domain_icon.ts [hunk 1]
```diff
@@ -44,6 +47,14 @@ export const domainIcon = (
     case "cover":
       return coverIcon(compareState, stateObj);
 
+    case "device_tracker":
+      switch (stateObj?.attributes.source_type) {
+        case "router":
+          return compareState === "home" ? mdiLanConnect : mdiLanDisconnect;
+        default:
+          return mdiAccount;
```
Comment (bramkragten) [inline src/common/entity/domain_icon.ts:55]: You should remove it from fixed domain icons now
<https://github.com/home-assistant/frontend/blob/dev/src/common/const.ts#L79>

Comment (Mariusthvdb) [reply src/common/entity/domain_icon.ts:55]: thx

File: src/common/entity/domain_icon.ts [hunk 2]
```diff
@@ -44,6 +47,14 @@ export const domainIcon = (
...
+      }
```
Comment (bramkragten) [inline src/common/entity/domain_icon.ts:56]:
```suggestion
      if (stateObj?.attributes.source_type === "router") {
          return compareState === "home" ? mdiLanConnect : mdiLanDisconnect;
      }
      return mdiAccount;
```

**Comment by Mariusthvdb** [comment]
cant we do this for domain switch too:
```
    case "switch":
      return state && state === "off" ? mdiToggleSwitchOff : mdiToggleSwitch;
```

**Comment by bramkragten** [comment]
This PR is not about switch, keep it about the PR please.

For formatting, run prettier :-)

**Review by bramkragten — APPROVED** [review]

**Pull request merged**